### PR TITLE
`azurerm_databricks_workspace` - add support for Customer Managed Keys for Managed Services

### PR DESCRIPTION
**Pull request by WodansSon**
Add support for new functionality exposed in the `2021-04-01-preview` API

### CODE REVIEW

**Comment by WodansSon** [comment]
image

**Comment by WodansSon** [comment]
image

File: azurerm/internal/services/databricks/databricks_customer_managed_key_resource_test.go [hunk 1]
```diff
@@ -151,7 +151,7 @@ resource "azurerm_databricks_workspace" "test" {
 }
 
 %[4]s
-`, data.RandomInteger, data.Locations.Primary, keyVault, cmk)
+`, data.RandomInteger, "eastus2", keyVault, cmk)
```
Comment (katbyte) [inline azurerm/internal/services/databricks/databricks_customer_managed_key_resource_test.go:154]: could we use the alt location? it is by default east2 in our ci

**Review by katbyte — APPROVED** [review]
Aside from one comment this LGTM 🍕

File: azurerm/internal/services/databricks/databricks_customer_managed_key_resource_test.go (hunk 1 above)
Comment (WodansSon) [reply azurerm/internal/services/databricks/databricks_customer_managed_key_resource_test.go:154]: fixed

**Comment by WodansSon** [comment]
image

**Pull request merged**